controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,144 @@
+    // Only one supervisor instance can exist in a single controller. As such the same one will be used
+    // every time
+    webots::Supervisor supervisor = webots::Supervisor();
+    for (int i = 2; i < argc; i++) {
+        // Load the Node of the robot by def argument and add to vector
+        webots::Node* tempNode = supervisor.getFromDef(argv[i]);
+        otherRobotsNodes.push_back(tempNode);
+    }
+
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::array<double, 4>>>();
+        minDistance       = config["minDistance"].as<double>();
+        zHeight           = config["zHeight"].as<double>();
+    }
+    catch (const YAML::BadFile& e) {
+        std::cerr << e.msg << std::endl;
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+
+    webots::Node* tempNode = supervisor.getFromDef("test");
+    xSize                  = tempNode->getField("xSize")->getSFFloat();
+    ySize                  = tempNode->getField("ySize")->getSFFloat();
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+
+    // Generate distributions for random number systems in the following loop
+    std::uniform_int_distribution<> xDistrib(0, xSize * 100);
+    std::uniform_int_distribution<> yDistrib(0, ySize * 100);
+    std::uniform_int_distribution<> rotDistrib(0, rotations.size() - 1);
+
+    while (supervisor.step(timeStep) != -1) {
+
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions = std::vector<std::array<double, 3>>();
+        // Loop through every robot
+        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
+            // Assume there is no collision
+            bool collision = false;
+            // newPos will be a "Proposed location" for a robot to teleport to
+            std::array<double, 3> newPos;
+            do {
+                collision = false;
+                // Generate a new random location
+                newPos[0] = xSize / 2 - xDistrib(gen) / 100;
+                newPos[1] = ySize / 2 - yDistrib(gen) / 100;
+                newPos[2] = zHeight;
+
+                // Loop through the vector of existing proposed locations and see if the new one is going to
+                // collide with any of them
+                for (std::array<double, 3> testPos : positions) {
+                    double distance = sqrt(pow((newPos[1] - testPos[1]), 2) + pow((newPos[0] - testPos[0]), 2));
```

Comment:
```suggestion
                    double distance = std::sqrt(std::pow((newPos[1] - testPos[1]), 2) + std::pow((newPos[0] - testPos[0]), 2));
```

Reply:
What's the purpose of specifying std here?

Reply:
Because it's part of the std library?

Reply:
You should always be prefixing your standard library functions with `std::` when you're doing them in c++

In this case in particular it's even more important as if you don't it uses the C version of the functions which will automatically promote the input variables to doubles as C doesn't have the concept of overloading.
If you use the `std::` versions it will use the type of input that you are using (so if you put in floats it will use the float version). Since in this case, you are already using doubles it probably doesn't make any practical difference but it's a habit you should be getting into anyway.

Reply:
The only other potential issue are other libraries defining these functions in the global namespace. We aren't using any libraries that do this horrendous practice, but as @TrentHouliston said, its a good habit to get into

Reply:
Gotcha. The weird back and forth between C++ using some C libraries still confuses me sometimes

Reply:
As a simple rule of thumb (this only applies to the C and C++ standard library header files).

If the header file that you are including has a `.h` on the end of it, it is a C standard library header file (e.g. `math.h` and `stdio.h`).

If the header file has no extension at all, it is a C++ standard library header file (e.g. `cmath` and `iostream`).

If you want to use a C standard library header file in a C++ program (e.g. you want the math functions), generally you drop the extension and put a `c` on the from (e.g. `math.h` becomes `cmath`, `stdlib.h` becomes `cstdlib`). Always worth looking at [en.cppreference.com](https://en.cppreference.com/w/cpp) to see what the correct header file to include is

Reply:
Amazing, thanks for that!

Reply:
Once you're no longer dealing with standard library headers though all bets are off...